### ISSUE TITLE
Code coverage not getting generated in headless mode. in 3.10.0 

### ISSUE THREAD

**Issue by omkar-nath** [cypress-io/code-coverage#611]
Hi, We are using cypress/code-coverage - 3.10.0 and Cypress 10.3.1. In interactive mode we are able to see code coverage but when run in headless mode. 

```
Cannot find coverage file /Users/omkar/<folder>/uitest/.nyc_output/out.json
Skipping coverage report
```
I have tried with --headed option also but the result is same.

**Comment by gonzalo-novak** [comment]
Hello Omkar,

I had this issue, and I saw we can get the coverage report by setting up the following environment variable.

```
cypress run --browser chrome -e codeCoverageTasksRegistered=true --spec 'cypress/e2e/all.cy.ts'
``` 

The `codeCoverageTasksRegistered` variable is set in the interactive mode but not in headless mode. So, we need to configure it manually.

**Comment by ojemuyiwa** [comment]
> Hello Omkar,
> 
> I had this issue, and I saw we can get the coverage report by setting up the following environment variable.
> 
> ```
> cypress run --browser chrome -e codeCoverageTasksRegistered=true --spec 'cypress/e2e/all.cy.ts'
> ```
> 
> The `codeCoverageTasksRegistered` variable is set in the interactive mode but not in headless mode. So, we need to configure it manually.

sadly I get `sh: -e: command not found` and setting it in cypress `env: { codeCoverageTasksRegistered: true }` doesn't work either

**Comment by gonzalo-novak** [comment]
> > Hello Omkar,
> > I had this issue, and I saw we can get the coverage report by setting up the following environment variable.
> > ```
> > cypress run --browser chrome -e codeCoverageTasksRegistered=true --spec 'cypress/e2e/all.cy.ts'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The `codeCoverageTasksRegistered` variable is set in the interactive mode but not in headless mode. So, we need to configure it manually.
> 
> sadly I get `sh: -e: command not found` and setting it in cypress `env: { codeCoverageTasksRegistered: true }` doesn't work either

If you run this command on your terminal, you may need to point to the cypress binary in the node_modules directory. Otherwise, your terminal might not find the cypress binary in the $PATH variable of your computer. Try to use that command by creating a script in the package.json. For instance:

```
"scripts": {
  "e2e:coverage": "cypress run --browser chrome -e codeCoverageTasksRegistered=true --spec 'cypress/e2e/<the_test_ file_you_want_to_run>'"
}
```

**Comment by ojemuyiwa** [comment]
For anyone experiencing similar the steps used to get this working headless in the pipeline for me included:
package.json using *"nyc": "^15.1.0"*
`"nyc": {
        "extends": "@istanbuljs/nyc-config-typescript",
        "all": true,
        "report-dir": "e2e/coverage",
        "temp-dir": "e2e/.nyc_output",
        "exclude": [
            "server",
            "e2e",
            "dist",
            "tmp",
            "mock-data",
            "coverage",
            "node_modules",
            "src/**/*.spec.ts",
            "src/**/*.spec.tsx"
        ],
        "instrument": false,
        "include": [
            "src/**/*.ts",
            "src/**/*.tsx"
        ]
    }`

*"@babel/preset-react": "^7.14.5"*
.babelrc *NOTE* default configuration included and just added *Istanbul*
` "env": {
        "e2e": {
            "presets": [
                [
                    "@nrwl/react/babel",
                    {
                        "runtime": "automatic"
                    }
                ]
            ],
            "plugins": [
                [
                    "import",
                    { "libraryName": "antd", "libraryDirectory": "lib" },
                    "antd"
                ],
                "istanbul"
            ]
        }
    }`

Using the documentation: https://docs.cypress.io/guides/tooling/code-coverage#Using-code-transpilation-pipeline
For Nx Cli environment ("nx": "15.9.2") for some reason the `transform-class-properties` does not work and asks for `babel-transform-class-properties` to be installed which broke the build so I left it out altogether.

**Comment by nagash77** [comment]
Hi @omkar-nath , I realize this issue is quite old, but did @ojemuyiwa 's suggestion address your problem?

**Comment by SparksFlyx3** [comment]
My team also experiences issues with generating Cypress code coverage reports in our Jenkins pipeline. It works totally fine locally with various formats (txt, json, cobertura, html). Did anybody find the cause and/or a solution for this?

**Comment by nagash77** [comment]
Hi @SparksFlyx3  have you tried the suggestion in [this comment](https://github.com/cypress-io/code-coverage/issues/611#issuecomment-1576653922)?  If that does not address your problem I would suggest either opening a new issue with a reproducible example or checking out the Cypress Discord Community Discord chat, it can be helpful for debugging or answering questions on how to use Cypress.

**Comment by DavidLGoldberg** [comment]
Same issue, CLI/headless not working the way the runner did with "@cypress/code-coverage"

Tried a bunch of this to no avail.

I went with:
`"cypress:run": "rm -rf ./coverage && rm -rf ./.nyc_output && cypress run <stuff>` # could use rimraf for portability of course.

Easy enough.  Wish I thought of it sooner.  Sort of seems safer and more foolproof anyway.  Removes a vector for misleading reports.

If anyone has a reason why this is a bad idea, I'd be curious!
